### DISCUSSION TITLE
Allow to overload operators for existing classes

### DISCUSSION THREAD

**Discussion by ygc369** [dotnet/csharplang#6729] [Language Ideas] [answered]
In C#, adding new member functions for existing classes has already been supported for a long time. But we still can't add new operator overloading for existing classes, as operator overloading uses static functions, not member functions. What's a pity! 
I want C# to support adding new static functions generally for existing classes, or at least support overloading operators for existing classes as a special case.
For example:
```C#
static class stringExtension extend string //use "extend" as a new keyword to specify which existing class to extend
{
    public static string operator*(string s, uint n) // new static function for the existing class to extend
    {
        string res = ""; uint i; for(i=0; i<n; i++) res+=s; return res;
    }
}
.......
int main(....)
{
    string str = "abc" * 3; //str="abcabcabc";
    ......
}
```

**Comment by HaloFour** [comment, marked as answer]
I believe that would be covered by #5498 which proposes a new syntax that would allow any type of extension member on an existing type.
